tw-tasks-core/src/main/java/com/transferwise/tasks/triggering/KafkaTasksExecutionTriggerer.java

```diff
@@ -398,22 +404,45 @@ protected void registerCommitException(String bucketId, Throwable t) {
     }
   }
 
-  private KafkaConsumer<String, String> initKafkaConsumer(String bucketId, BucketProperties bucketProperties) {
-    String groupId = (String) kafkaConsumerProps.get(ConsumerConfig.GROUP_ID_CONFIG);
-    if (groupId == null) {
-      throw new IllegalStateException("Kafka consumer group id is not set.");
-    }
+  private KafkaProducer<String, String> createKafkaProducer() {
+    Map<String, Object> configs = new HashMap<>();
+    configs.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, tasksProperties.getTriggering().getKafka().getBootStrapServers());
+
+    configs.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+    configs.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+
+    configs.put(ProducerConfig.ACKS_CONFIG, "all");
+    configs.put(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, 1);
+    configs.put(ProducerConfig.MAX_BLOCK_MS_CONFIG, "5000");
+    configs.put(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG, "5000");
+    configs.put(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, "10000");
+    configs.put(ProducerConfig.LINGER_MS_CONFIG, "5");
+
+    configs.putAll(tasksProperties.getTriggering().getKafka().getProperties());
+
+    KafkaProducer<String, String> kafkaProducer = new KafkaProducer<>(configs);
+
+    return kafkaProducer;
+  }
+
+  private KafkaConsumer<String, String> createKafkaConsumer(String bucketId, BucketProperties bucketProperties) {
+    String groupId = tasksProperties.getGroupId();
 
     if (bucketProperties.getTriggerSameTaskInAllNodes()) {
       log.info("Using same task triggering on all nodes strategy for bucket '{}'.", bucketId);
       groupId += "." + tasksProperties.getClientId();
     }
 
-    Map<String, Object> configs = new HashMap<>(kafkaConsumerProps);
+    Map<String, Object> configs = new HashMap<>();
+    configs.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, tasksProperties.getTriggering().getKafka().getBootStrapServers());
     configs.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, bucketProperties.getTriggersFetchSize());
     configs.put(ConsumerConfig.GROUP_ID_CONFIG, groupId);
     configs.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
-    configs.put(ConsumerConfig.CLIENT_ID_CONFIG, configs.getOrDefault(ConsumerConfig.CLIENT_ID_CONFIG, "") + ".tw-tasks.bucket." + bucketId);
```

Comment:
Won't it be helpful to have bucket id included? Also if I remember correctly, if there are 2 consumers with the same id some warnings like 
```
javax.management.InstanceAlreadyExistsException: kafka.consumer:type=app-info,id=clientId
```
might start appearing during startup